DependencyInjection/FOSMessageExtension.php

```diff
@@ -51,10 +52,34 @@ public function load(array $configs, ContainerBuilder $container)
         $container->setAlias('fos_message.spam_detector', $config['spam_detector']);
         $container->setAlias('fos_message.twig_extension', $config['twig_extension']);
 
-        $container->setAlias('fos_message.new_thread_form.type', $config['new_thread_form']['type']);
+        // BC management
+        $newThreadFormType = $config['new_thread_form']['type'];
+        $replyFormType = $config['reply_form']['type'];
+
+        if (!class_exists($newThreadFormType)) {
+            @trigger_error('Using a service reference in configuration key "fos_message.new_thread_form.type" is deprecated since version 1.2 and will be removed in 2.0. Use the class name of your form type instead.', E_USER_DEPRECATED);
+
+            // Old syntax (service reference)
+            $container->setAlias('fos_message.new_thread_form.type', $newThreadFormType);
+        } else {
+            // New syntax (class name)
+            $container->getDefinition('fos_message.new_thread_form.factory.default')
+                ->replaceArgument(1, $newThreadFormType);
+        }
+
+        if (!class_exists($replyFormType)) {
```

Comment:
I find this more readable than a foreach, as there would be a concatenated variable everywhere.